Symfony/CS/Tests/Tokenizer/TokenTest.php

```diff
@@ -240,4 +240,101 @@ public function testPropertiesOfNonArrayToken()
         $this->assertNull($token->getLine());
         $this->assertFalse($token->isArray());
     }
+
+    /**
+     * @dataProvider provideEquals
+     */
+    public function testEquals($token, $equals, $other, $caseSensitive = null)
+    {
+        if (!isset($caseSensitive)) {
+            $this->assertSame($equals, $token->equals($other));
+        } else {
+            $this->assertSame($equals, $token->equals($other, $caseSensitive));
+        }
+    }
+
+    public function provideEquals()
+    {
+        return array(
+            array($this->getBraceToken(), false, '!'),
+            array($this->getBraceToken(), false, '!', true),
+            array($this->getBraceToken(), false, '!', false),
+            array($this->getBraceToken(), true, '('),
+            array($this->getBraceToken(), true, '(', true),
+            array($this->getBraceToken(), true, '(', false),
+            array(new Token(array(T_FUNCTION, 'function', 1)), false, '('),
```

Comment:
please store that token in variable, it will be more readable (and use less memory btw)